Checker.cpp

```diff
@@ -1,22 +1,107 @@
 #include <assert.h>
 #include <iostream>
+
 using namespace std;
 
-bool batteryIsOk(float temperature, float soc, float chargeRate) {
-  if(temperature < 0 || temperature > 45) {
-    cout << "Temperature out of range!\n";
-    return false;
-  } else if(soc < 20 || soc > 80) {
-    cout << "State of Charge out of range!\n";
-    return false;
-  } else if(chargeRate > 0.8) {
-    cout << "Charge Rate out of range!\n";
-    return false;
-  }
-  return true;
+
+class BatteryManagementSystem {
+private:
+    static constexpr float kMinTemperature = 0.0f;
+    static constexpr float kMaxTemperature = 45.0f;
+    static constexpr float kMinSoc = 20.0f;
+    static constexpr float kMaxSoc = 80.0f;
+    static constexpr float kMaxChargeRate = 0.8f;
+
+public:
+    bool isTemperatureInRange(float temperature) const;
+    bool isSocInRange(float soc) const;
+    bool isChargeRateInRange(float chargeRate) const;
```

Comment:
mark all the parameter check methods as static! because class does not contain any data member and none of the functions mutate any state of the object and fundamental question is : why does these method need "this" argument?